<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Assembly Completeness and Quality</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomics Workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Getting_Started.html">Scan Raw Data</a>
</li>
<li>
  <a href="Trim_Adaptors.html">Trim Adaptors</a>
</li>
<li>
  <a href="Create_Histogram.html">Create Histogram</a>
</li>
<li>
  <a href="Assemble_With_Various_Data_Types.html">Assemble with Various Data Types</a>
</li>
<li>
  <a href="Assembly_Statistics.html">Assembly Statistics</a>
</li>
<li>
  <a href="Assess_Completeness.html">Assembly Completeness and Quality</a>
</li>
<li>
  <a href="Mask_Repetitive_Elements.html">Mask Repetitive Elements</a>
</li>
<li>
  <a href="Gene_Prediction_Begin_Annotation.html">Gene Prediction &amp; Begin Annotation</a>
</li>
<li>
  <a href="Call_Variants.html">Call Variants</a>
</li>
<li>
  <a href="Glossary.html">Glossary</a>
</li>
<li>
  <a href="Unix_Linux_Cheat_Sheet.html">Common Unix / Linux Commands</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Assembly Completeness and Quality</h1>

</div>


<div id="busco" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Busco</h1>
<p><strong>Busco</strong> is a software that assesses genome completeness based on evolutionarily-informed expectations of gene content of near-universal single-copy orthologs by searching for these select sets of orthologous genes. There are several databases that can be used with Busco, and can be downloaded <a href="https://busco.ezlab.org">here</a>.</p>
<p>Busco can be run with or without internet, so we will consider two ways to run this software:</p>
<p><strong>Option 1:</strong>
For option 1, we will assume that you have internet connection on the node you are running your job on.
Note: Busco does not have the ability to direct an output file to a different folder than the folder you submit the job from, even with a path identified. In this case, consider creating a Busco folder.</p>
<pre class="bash"><code>#make directory / folder
mkdir Busco
#
#change directory to your Busco folder
cd Busco</code></pre>
<p>Be sure that it is available and loaded on your HPC, and write and submit a job with the following code:</p>
<pre class="bash"><code>echo + `date` job $JOB_NAME started in $QUEUE with jobID=$JOB_ID on $HOSTNAME
echo + NSLOTS = $NSLOTS
#
busco  -o output_file -i /path/to/input/fasta/file.fasta -l mammalia_odb10 -c $NSLOTS -m genome
#
echo = `date` job $JOB_NAME done
#
#Explanation:
#-o: name of the output folder and files
#-i: input file (FASTA)
#-l: lineage - ex. mammalian / name of the database (This will automatically connected and download the database from the BUSCO website)
#-c: number of CPUs
#-m: mode (options are *genome*, transcriptome, proteins)</code></pre>
<p><strong>Option 2:</strong>
If you do not have internet connection on the node you are running the software on, you can download the database and run the program offline.</p>
<pre class="bash"><code>#make directory / folder
mkdir Busco
#
#change directory to your Busco folder
cd Busco
#
#download database from web, ex. mammal database
wget https://busco-data.ezlab.org/v5/data/lineages/mammalia_odb10.2021-02-19.tar.gz
#
#extract files within Busco folder
tar -zxf mammalia_odb10.2021-02-19.tar.gz</code></pre>
<p>Write and submit a job with the following code:</p>
<pre class="bash"><code>echo + `date` job $JOB_NAME started in $QUEUE with jobID=$JOB_ID on $HOSTNAME
echo + NSLOTS = $NSLOTS
#
busco  -o output_file -i /path/to/input/fasta/file.fasta -l mammalia_odb10 -c $NSLOTS -m genome --offline --download_path /path/to/datasets
#
echo = `date` job $JOB_NAME done
#
#Explanation:
#-o: name of the output folder and files
#-i: input file (FASTA)
#-l: lineage - ex. mammalian / name of the database (This will automatically connected and download the database from the BUSCO website)
#-c: number of CPUs
#-m: mode (options are *genome*, transcriptome, proteins)</code></pre>
</div>
<div id="blobtools" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Blobtools</h1>
<p><strong>Blobtools</strong> is a command line tool designed for interactive quality assessment of genome assemblies and contaminant detection and filtering. We will use both Blobtools and Blobtools2 in the following steps:</p>
<p><strong>1)</strong> Apply <a href="https://github.com/DRL/blobtools">Blobtools</a></p>
<p><strong>2)</strong> Create <a href="https://github.com/blobtoolkit/blobtoolkit">Blobtools2</a> database</p>
<p><strong>3)</strong> Add data to database</p>
<p><strong>4)</strong> Create interactive html page with results</p>
<div id="apply-blobtools" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Apply Blobtools</h2>
<p>Note: Identify the version of Blobtools you are using when loading the module on your HPC
Write and submit a job script with the following code:</p>
<pre class="bash"><code>echo + `date` job $JOB_NAME started in $QUEUE with jobID=$JOB_ID on $HOSTNAME
echo + NSLOTS = $NSLOTS
#
blastn -db /data/genomics/db/ncbi/db/latest_v4/nt/nt -query /path/to_assembly/file.fasta -outfmt &quot;6 qseqid staxids bitscore std&quot; -max_target_seqs 20 -max_hsps 1 -evalue 1e-20 -num_threads $NSLOTS -out clouded_leopard_blast.out
#
echo = `date` job $JOB_NAME done
#
#Explanation
#-db: ncbi nucleotide database
#-query: input file (FASTA)
#-outfmt: format of the output file (important to for blobtools) 
#-max_target_seqs: Number of aligned sequences to keep.
#-max_hsps: Maximum number of HSPs (alignments) to keep for any single query-subject pair.
#-num_threads: number of CPUs
#-out: name of the output file</code></pre>
</div>
<div id="create-blobtools2-database" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Create Blobtools2 Database</h2>
<p>The following can be run on an interactive, or computational node. First, load blobtools, specifying the version you are using. Next, submit the following code on your interactive node:</p>
<pre class="bash"><code>blobtools create --fasta /path/to_assembly/file.fasta species_blobt_nb
#
#Explanation
#create: command to create the database with blobtools2
#--fasta: path and name of the assembly fasta file
#species_blobt_nb: name of the database</code></pre>
</div>
<div id="add-data-to-database" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Add Data to Database</h2>
<p>Once you have a Blob database, additional data can be added by parsing analysis output files into one or more fields using the blobtools “add” command. Preset parsers are available for a range of analysis types, such as BLAST or “Diamond hits” with taxonomic assignments for scaffolds or contigs. Read mapping files provide base and read coverage and BUSCO results show completeness metrics for the genome assembly.</p>
<p>This step can also be run on an interactive node:</p>
<pre class="bash"><code>blobtools add --threads 3 --hits species_blast.out --taxrule bestsumorder --taxdump /path/to/taxdump/folder species_blobt_nb 
#
blobtools add --threads 3 --cov species_sorted.bam species_blobt_nb 
#
blobtools add --threads 3 --busco table.tsv species_blobt_nb</code></pre>
</div>
<div id="create-interactive-html-page-with-results" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Create Interactive HTML Page with Results</h2>
<p>After you finish creating and adding data to the database in order to visualize the results, install blobtools2 on your personal machine and download the database folder.</p>
<pre class="bash"><code>#tar zip folder
tar -czvf name-of-archive.tar.gz /path/to/directory/or/file

#download using ffsend
module load bio/ffsend

#ffsend
ffsend upload &lt;file&gt;

#Install blobtools on your machine using conda

#make new folder for blobtool results
mkdir blobtools_results

#move to new folder
cd blobtools_results

#move or ffsend folder to this folder

#unzip file
untar -xzvf archive.tar.gz

#After installing blobtools2, activate your conda environment and from the program folder by running the following command on your database folder:
conda activate btk_env
./blobtools view --remote path/to/species_nb</code></pre>
<p>You should be able to interact with and visualize results</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
